### PR TITLE
Make new blip compatible with wdio testing

### PR DESCRIPTION
**Pull request by franck-fourel**

### CODE REVIEW

File: packages/yourloops/pages/signup/signup.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ function SignUpPage(): JSX.Element {
       >
         <Grid item xs={12}>
           <SignUpFormStateProvider>
-            <Card className={classes.Card}>
+            <Card id="signup" className={classes.Card}>
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup.tsx:85]:
```suggestion
            <Card id="signup-card" className={classes.Card}>
```

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 2]
```diff
@@ -267,6 +267,7 @@ function SignUpProfileForm(props: SignUpFormProps): JSX.Element {
       />
       <div id="signup-profileform-button-group">
         <Button
+          id="signup-steppers-back"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:270]:
```suggestion
          id="button-signup-steppers-back"

```

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 3]
```diff
@@ -275,6 +276,7 @@ function SignUpProfileForm(props: SignUpFormProps): JSX.Element {
           {t("signup-steppers-back")}
         </Button>
         <Button
+          id="signup-steppers-next"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:279]:
```suggestion
          id="button-signup-steppers-next"

```

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 4]
```diff
@@ -175,7 +175,7 @@ function SignUpProfileForm(props: SignUpFormProps): JSX.Element {
         id="firstname"
         className={classes.TextField}
         margin="normal"
-        label={t("firstname")}
+        label={t("firstName")}
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:178]:
```suggestion
        label={t("firstname")}
```
Yourloops tr key is in lowercase

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 5]
```diff
@@ -188,7 +188,7 @@ function SignUpProfileForm(props: SignUpFormProps): JSX.Element {
         id="lastname"
         className={classes.TextField}
         margin="normal"
-        label={t("lastname")}
+        label={t("lastName")}
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:191]:
```suggestion
        label={t("lastname")}

```
Yourloops tr key is in lowercase

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 6]
```diff
@@ -138,10 +140,12 @@ export default function SignUpConsent(props: SignUpFormProps): JSX.Element {
           )}
         />
         <FormControlLabel
+          id="signup-consent-terms-ctl"
           className={classes.FormControlLabel}
           control={
             <Checkbox
-              checked={state.formValues.terms}
+            id="signup-consent-terms-check"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-consent.tsx:147]:
```suggestion
            id="checkbox-signup-consent-terms"

```

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 7]
```diff
@@ -151,12 +155,13 @@ export default function SignUpConsent(props: SignUpFormProps): JSX.Element {
           }
           label={t(
             `signup-consent-${state.formValues.accountRole}-terms-condition`,
-            { terms: "Terms or Use" }
+            { terms: "Terms of Use" }
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-consent.tsx:158]:
```suggestion
            { terms: t("Terms of Use") }
```
???

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 8]
```diff
@@ -151,12 +155,13 @@ export default function SignUpConsent(props: SignUpFormProps): JSX.Element {
...
           label={t(
             `signup-consent-${state.formValues.accountRole}-terms-condition`,
-            { terms: "Terms or Use" }
+            { terms: "Terms of Use" }
           )}
           labelPlacement="end"
         />
         <div id="signup-consent-button-group">
           <Button
+            id="signup-steppers-back"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-consent.tsx:164]:
```suggestion
            id="button-signup-steppers-back"

```

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 9]
```diff
@@ -166,6 +171,7 @@ export default function SignUpConsent(props: SignUpFormProps): JSX.Element {
             {t("signup-steppers-back")}
           </Button>
           <Button
+            id="signup-steppers-next"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-consent.tsx:174]:
```suggestion
            id="button-signup-steppers-next"

```

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 10]
```diff
@@ -170,6 +174,7 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
           {t("signup-steppers-back")}
         </Button>
         <Button
+          id="signup-steppers-next"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:177]:
```suggestion
          id="button-signup-steppers-next"

```

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 11]
```diff
@@ -162,6 +165,7 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
       </FormControl>
       <div id="signup-account-selector-button-group">
         <Button
+          id="signup-steppers-back"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:168]:
```suggestion
          id="button-signup-steppers-back"

```

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 12]
```diff
@@ -149,6 +151,7 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
             <FormControlLabel
               className={classes.FormControlLabel}
               value="caregiver"
+              id="signup-account-caregiver"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:154]:
```suggestion
              id="form-label-signup-account-caregiver"

```

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 13]
```diff
@@ -136,6 +137,7 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
             <FormControlLabel
               className={classes.FormControlLabel}
               value="hcp"
+              id="signup-account-hcp"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:140]:
```suggestion
              id="form-label-signup-account-hcp"

```

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 14]
```diff
@@ -112,6 +112,7 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
 
   return (
     <form
+      id="signup-form"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:115]:
```suggestion
      id="form-signup"

```
??

File: packages/yourloops/pages/signup/signup-account-form.tsx [hunk 15]
```diff
@@ -237,6 +238,7 @@ function SignUpAccountForm(props: SignUpFormProps): JSX.Element {
           {t("signup-steppers-back")}
         </Button>
         <Button
+          id="signup-steppers-create"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-form.tsx:241]:
```suggestion
          id="button-signup-steppers-create"

```

File: packages/yourloops/pages/signup/signup-account-form.tsx [hunk 16]
```diff
@@ -229,6 +229,7 @@ function SignUpAccountForm(props: SignUpFormProps): JSX.Element {
       />
       <div id="signup-accountform-button-group" className={classes.Buttons}>
         <Button
+          id="signup-steppers-back"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-form.tsx:232]:
```suggestion
          id="button-signup-steppers-back"

```

File: packages/yourloops/pages/profile/profile.tsx [hunk 17]
```diff
@@ -399,10 +401,16 @@ export const ProfilePage: FunctionComponent = () => {
...
+            <Button
+              id="cancel-profile"
+              variant="contained"
+              color="secondary"
+              onClick={onCancel}
+              className={classes.button}>
               {t("common-cancel")}
             </Button>
             <Button
+              id="save-profile"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/profile/profile.tsx:413]:
```suggestion
              id="button-save-profile"
```

File: packages/yourloops/pages/profile/profile.tsx [hunk 18]
```diff
@@ -399,10 +401,16 @@ export const ProfilePage: FunctionComponent = () => {
             </Select>
           </FormControl>
           <div style={{ display: "flex", justifyContent: "flex-end" }}>
-            <Button variant="contained" color="secondary" onClick={onCancel} className={classes.button}>
+            <Button
+              id="cancel-profile"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/profile/profile.tsx:405]:
```suggestion
              id="button-cancel-profile"
```

File: packages/yourloops/pages/login/login.tsx [hunk 19]
```diff
@@ -208,7 +209,7 @@ function Login(props: RouteComponentProps): JSX.Element {
                   }}
                 />
               </form>
-              <Link component={RouterLink} to="/request-password-reset">
+              <Link id="password-reset" component={RouterLink} to="/request-password-reset">
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:212]:
```suggestion
              <Link id="link-password-reset" component={RouterLink} to="/request-password-reset">
```

File: packages/yourloops/pages/login/login.tsx [hunk 20]
```diff
@@ -167,6 +167,7 @@ function Login(props: RouteComponentProps): JSX.Element {
             </CardMedia>
             <CardContent className={classes.CardContent}>
               <form
+                id="login-form"
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:170]:
```suggestion
                id="form-login"
```

File: packages/yourloops/components/header-bar.tsx [hunk 21]
```diff
@@ -151,8 +155,16 @@ function HeaderBar(props: HeaderProps): JSX.Element {
           }}
           open={open}
           onClose={handleClose}>
-          <MenuItem onClick={handleOpenProfilePage}>{t("menu-account-preferences")}</MenuItem>
-          <MenuItem onClick={handleLogout}>{t("Logout")}</MenuItem>
+          <MenuItem
+            id="open-profile"
```
Comment (yvon-dblg) [inline packages/yourloops/components/header-bar.tsx:159]:
```suggestion
            id="menuitem-open-profile"
```

File: packages/yourloops/components/header-bar.tsx [hunk 22]
```diff
@@ -121,15 +121,19 @@ function HeaderBar(props: HeaderProps): JSX.Element {
...
       <div className={classes.accountMenu}>
         <div className={classes.accountInfos}>
-          <div className={classes.accountName}>{`${user?.profile?.firstName} ${user?.profile?.lastName}`}</div>
+          <div className={classes.accountName} id="appbar-account-name">
+            {name}
+          </div>
           <div className={classes.accountType}>{role}</div>
         </div>
         <IconButton
+          id="menu-appbar-button"
```
Comment (yvon-dblg) [inline packages/yourloops/components/header-bar.tsx:136]:
```suggestion
          id="appbar-menu-button-account"
```

File: packages/blip/app/components/patient-data.js [hunk 23]
```diff
@@ -217,7 +217,7 @@ class PatientDataPage extends React.Component {
         loader = <Loader />;
         break;
       default:
-        errorDisplay = <p>{errorMessage ?? t('Failed somewhere')}</p>;
+        errorDisplay = <p id="display-message">{errorMessage ?? t('Failed somewhere')}</p>;
```
Comment (yvon-dblg) [inline packages/blip/app/components/patient-data.js:220]:
```suggestion
        errorDisplay = <p id="loading-error-message">{errorMessage ?? t('Failed somewhere')}</p>;
```

**Review by yvon-dblg — COMMENTED** [review]
Just some naming suggestions.

File: packages/blip/app/components/patient-data.js (hunk 23 above)
Comment (franck-fourel) [reply packages/blip/app/components/patient-data.js:220]: I don't see the benefit of specializing so much this kind of id

File: packages/yourloops/pages/signup/signup.tsx (hunk 1 above)
Comment (franck-fourel) [reply packages/yourloops/pages/signup/signup.tsx:85]: this one should be 
```suggestion
            <Card id="card-signup" className={classes.Card}>
```

File: packages/blip/app/components/patient-data.js (hunk 23 above)
Comment (yvon-dblg) [reply packages/blip/app/components/patient-data.js]: Just to be sure it is an unique id. If too generic, we may ends up having 2x the same id at the same time.

File: packages/yourloops/pages/profile/profile.tsx [hunk 24]
```diff
@@ -307,7 +308,9 @@ export const ProfilePage: FunctionComponent = () => {
       <Snackbar params={snackbarParams} />
       <Container className={classes.container} maxWidth="sm">
         <div style={{ display: "flex", flexDirection: "column", margin: "16px" }}>
-          <DialogTitle className={classes.title}>{t("account-preferences-title")}</DialogTitle>
+          <div className={classes.title} id="account-preferences-title">
```
Comment (yvon-dblg) [inline packages/yourloops/pages/profile/profile.tsx:311]: No need to change the `<DialogTitle/>` with a `<div/>` here, you can put the id in it.

File: packages/yourloops/pages/signup/signup-account-form.tsx [hunk 25]
```diff
@@ -246,6 +246,8 @@ function SignUpAccountForm(props: SignUpFormProps): JSX.Element {
       />
       <div id="signup-accountform-button-group" className={classes.Buttons}>
         <Button
+          id="button-signup-steppers-back"
+
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-form.tsx:250]:
```suggestion
```
Remove empty line

File: packages/yourloops/pages/signup/signup-account-form.tsx [hunk 26]
```diff
@@ -254,6 +256,8 @@ function SignUpAccountForm(props: SignUpFormProps): JSX.Element {
           {t("signup-steppers-back")}
         </Button>
         <Button
+          id="button-signup-steppers-create"
+
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-form.tsx:260]:
```suggestion
```
Remove empty line

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 27]
```diff
@@ -162,6 +165,8 @@ function SignUpAccountSelector(props: SignUpFormProps): JSX.Element {
       </FormControl>
       <div id="signup-account-selector-button-group">
         <Button
+          id="button-signup-steppers-back"
+
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:169]:
```suggestion
```
Remove empty line

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 28]
```diff
@@ -151,12 +155,13 @@ export default function SignUpConsent(props: SignUpFormProps): JSX.Element {
           }
           label={t(
             `signup-consent-${state.formValues.accountRole}-terms-condition`,
-            { terms: "Terms or Use" }
+            { terms: t("Terms of Use") }
```
Comment (yvon-dblg) [inline packages/yourloops/pages/signup/signup-consent.tsx:158]:
```suggestion
            { terms: t("terms-and-conditions") }
```
They key name is not good, but the translation is.

File: packages/yourloops/components/language-select.tsx [hunk 29]
```diff
@@ -47,16 +47,17 @@ function LanguageSelect(): JSX.Element {
...
+        <MenuItem id={lang} key={lang} value={lang}>
           {language}
         </MenuItem>
       );
     }
   }
 
   return (
-    <FormControl>
+    <FormControl >
```
Comment (yvon-dblg) [inline packages/yourloops/components/language-select.tsx:58]:
```suggestion
    <FormControl>
```
Avoid a useless change will be helpful for the potentials merge conflicts in others PR.

File: packages/yourloops/components/primary-header-bar.tsx [hunk 30]
```diff
@@ -129,7 +129,9 @@ function HeaderBar(props: HeaderProps): JSX.Element {
 
   let accountMenu = null;
   if (auth.isLoggedIn()) {
-    const user = auth.user;
+    const user: User = auth.user as User;
+    const name = `${getUserFirstName(user)} ${getUserLastName(user)}`;
```
Comment (yvon-dblg) [inline packages/yourloops/components/primary-header-bar.tsx:133]:
```suggestion
    const name = t("user-name", { firstName: getUserFirstName(user), lastName: getUserLastName(user) });
```
Key used for that case. In some languages, the order is not firstName + lastName, but the opposite.

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
Few things lefts I think.

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**